src/nvim/ex_cmds.c

```diff
@@ -2861,12 +2861,14 @@ void ex_append(exarg_T *eap)
     /* Look for the "." after automatic indent. */
     vcol = 0;
     for (p = theline; indent > vcol; ++p) {
-      if (*p == ' ')
+      if (*p == ' ') {
         ++vcol;
-      else if (*p == TAB)
+      } else if (*p == TAB) {
         vcol += 8 - vcol % 8;
-      else
+      } else {
+	update_screen(0);
```

Comment:
we try to avoid direct calls to `update_screen` if possible. would `redraw_later(curwin, VALID)` work?

Reply:
No. Screen does not redraw,if replaced `update_screen` to `redraw_later(curwin, VALID)`

Reply:
buffer content changed, shouldn't it be NOT_VALID instead ?

Reply:
That redraws everything which is wasteful. I would try `redrawWinline(wp, line)` first.

Reply:
Fixed. Deleted `update_screen()` and inserted `redrawWinline(curwin,line)` to other place.

Reply:
What about a for loop,that goes trough every line. Would that be okay in terms of resources?

Reply:
Fixed. I inserted `redrawWinline(curwin,lnum)` a bit too far. Now its fixed using `redrawWinline(curwin,lnum)`